readthedocs/projects/tasks.py

```diff
@@ -238,6 +234,7 @@ def run(self, version_pk):  # pylint: disable=arguments-differ
         :rtype: bool
         """
         try:
+            before_vcs.send(sender=self.version)
```

Comment:
This should be below `self.project = ...`

Reply:
ha, true